lib/Controller/Layout.php

```diff
@@ -726,6 +733,83 @@ function unretire($layoutId)
         ]);
     }
 
+    /**
+     * Set Enable Stats Collection of a layout
+     * @param int $layoutId
+     *
+     * @SWG\Put(
+     *  path="/layout/setenablestat/{layoutId}",
+     *  operationId="layoutSetEnableStat",
+     *  tags={"layout"},
+     *  summary="Enable Stats Collection",
+     *  description="Set Enable Stats Collection? to use for the collection of Proof of Play statistics for a Layout.",
+     *  @SWG\Parameter(
+     *      name="layoutId",
+     *      in="path",
+     *      description="The Layout ID",
+     *      type="integer",
+     *      required=true
+     *   ),
+     *  @SWG\Parameter(
+     *      name="enableStat",
+     *      in="formData",
+     *      description="Flag indicating whether the Layout stat is enabled",
+     *      type="integer",
+     *      required=true
+     *   ),
+     *  @SWG\Response(
+     *      response=204,
+     *      description="successful operation"
+     *  )
+     * )
+     *
+     * @throws XiboException
+     */
+    function setEnableStat($layoutId)
+    {
+        $layout = $this->layoutFactory->getById($layoutId);
+
+        if (!$this->getUser()->checkEditable($layout))
+            throw new AccessDeniedException(__('You do not have permissions to edit this layout'));
+
+        // Make sure we're not a draft
+        if ($layout->isChild())
+            throw new InvalidArgumentException(__('Cannot modify Layout from its Draft'), 'layoutId');
+
+        $enableStat = $this->getSanitizer()->getCheckbox('enableStat');
+
+        $layout->enableStat = $enableStat;
+        $layout->save();
+
+        // Return
+        $this->getState()->hydrate([
+            'httpStatus' => 204,
+            'message' => sprintf(__('For Layout %s Enable Stats Collection is set to %s'), $layout->layout, ($layout->enableStat == 1) ? 'On': 'Off')
```

Comment:
Please can we translate On and Off strings too?

Reply:
`sprintf(__('For Layout %s Enable Stats Collection is set to %s'), $layout->layout, ($layout->enableStat == 1) ? __('On') : __('Off'))` updated